src/seedu/addressbook/data/person/Address.java

```diff
@@ -7,52 +7,215 @@
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
+	
+	private boolean isPrivate;
+	public Block block;
+	public Street street;
+	public Unit unit;
+	public PostalCode postalCode;
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must follow the format: <BLOCK>, <STREET>, #<FLOOR>-<ROOM>, <POSTALCODE>, where BLOCK, FLOOR, ROOM are integers, and POSTALCODE is a 6 digit integer.";
 
     /**
      * Validates given address.
-     *
+     * 
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
+    	if (!isValidAddress(address)) {
+    		throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+    	}
+    	
+    	this.isPrivate = isPrivate;
+    	splitAddressAndSave(address);
     }
-
+    
+    /**
+     * Splits a valid address into its parts, and assigns them to the object instance.
+     * 
+     * @param address
+     */
+    private void splitAddressAndSave(String address) {
+    	String[] parts = splitAddress(address);
+    	String[] unitParts = splitUnitNumber(parts[2]);
+    	
+    	block = new Block(parts[0]);
+    	street = new Street(parts[1]);
+    	unit = new Unit(unitParts[0], unitParts[1]);
+    	postalCode = new PostalCode(parts[3]);
+    }
+    
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    	String[] parts = splitAddress(test);
+        
+        // Check if we have the correct number of parts
+        if (parts.length != 4)
+        	return false;
+        
+        // Check if unit number must begin with #
+        if (parts[2].charAt(0) != '#')
+        	return false;
+        
```

Comment:
assign literals such as '4' and the '#' symbol to constants with meaningful names so that someone seeing your code for the first time understands why you are checking against those numbers and symbols.